test/middleware/test-errors.js

```diff
@@ -0,0 +1,43 @@
+const { assert } = require('chai');
+const app = require('express')();
+const config = require('../../app/config');
+const log = require('bole')('middleware');
+const errors_middleware = require('../../app/middleware/errors')(app, config, log);
+
+
+describe('When a "state mismatch" error occurs', () => {
+  it('redirects the user to /login', () => {
+    let spy_session_destroyed = false;
+    let spy_cookie_cleared = 'NOT CLEARED';
+    let spy_redirected_to = 'NOT REDIRECTED';
+    let spy_next_called = false;
+
+    const err = Error('state mismatch, could not find [...]');
+    const req = {
+      session: {
+        destroy: (after_destroy) => {
+          spy_session_destroyed = true;
+          after_destroy();
+        },
+      },
+    };
+    const res = {
+      clearCookie: (name) => {
+        spy_cookie_cleared = name;
+      },
+      redirect: (url) => {
+        spy_redirected_to = url;
+      },
+    };
+    const next = () => {
```

Comment:
We already have a dependency on [`chai-spies`](https://github.com/chaijs/chai-spies), so you could use that to make this code simpler.

Reply:
I'm not familiar with it, I'll give it a play to see if it makes things more complicated.